### PR TITLE
Email

### PR DESCRIPTION
**Pull request by dooma**
Add email for grades and announcements.

I should comment my code :bangbang:

### CODE REVIEW

File: app/mailers/announce_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  # Send email to users from classroom when is posted a new announcement
+  #
+  # param [Hash] announcement, with email address and details
+  def new_announce_email(announcement)
+      announcement[:recipient].members.each do |user|
+        unless user.email == announcement[:recipient].owner.email
+          @announcement = announcement
+          @user = user
+          mail(:to => user.email,
+              :subject => "New announcement in #{announcement[:recipient][:title]}")
```
Comment (stas) [inline app/mailers/announce_mailer.rb]: User-end strings should be localized. We are using `_('string')` to localize text.

File: app/views/announce_mailer/new_announce_email.html.haml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+%h3
+  = _('Hello %s,') % @user.name
+
+%p
+  = _('An announcement was posted in the %s by %s') % [@announcement[:recipient][:title],@announcement[:recipient][:owner]]
+%p
```
Comment (stas) [inline app/views/announce_mailer/new_announce_email.html.haml]: Consider using a `blockquote` tag here to emphasize you are quoting the teacher.
<del>Also this should be checked if it is not `blank?` first.</del> Sorry was thinking about grades here, ignore this part.

File: app/views/grades_mailer/new_grade_email.html.haml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+%h3
+  = _("%s, graded you") % @grade_user.name
+
+%p
+  Informations about grade:
+  = _("Assignment: %s") % @content.assignment.title
+  = _("Lecture: %s") % @content.assignment.lecture.title
+  = _("Comment: %s") % @content.comment
```
Comment (stas) [inline app/views/grades_mailer/new_grade_email.html.haml]: This should be checked if it is not `blank?` first.

**Comment by stas** [comment]
Thanks, could you take a look at the notes I left.

**Comment by dooma** [comment]
You can merge this if you think it's ok. I will come with the email panel, but I have an issue with form for unlimited checkboxes, so I hope to push tomorrow.

**Comment by stas** [comment]
https://github.com/stas/coursewa.re/compare/0710499ea2fde97813498ccbff50a0d60613673a...6a7a9295d51866bcdee0cdbeae6ffaf88347b83f

I did some changes to mailers, some things to note:
- it's totally fine if you use longer names for variables, in ex.: `AnnounceMailer` vs `AnnounementMailer`
- I need to publish/adopt a coding style guide, so we all know we should use 2 spaces indentation and single quotes for string variables
- consider splitting pull requests to make them smaller next time (say one for announcement and another one for grades mailers)

Overall not bad at all, carry on!
I think we should do some skype calls in weekend to discuss together some next steps decisions in development as a team.

**Pull request merged**